main.go

```diff
@@ -115,8 +138,60 @@ func main() {
 	}
 
 	setupLog.Info("starting manager")
-	if err := mgr.Start(ctrl.SetupSignalHandler()); err != nil {
+	if err := mgr.Start(ctx); err != nil {
 		setupLog.Error(err, "problem running manager")
 		os.Exit(1)
 	}
 }
+
+// Tee consumes the received channel and mirrors the messages into 2 new channels.
+func tee[T any](ctx context.Context, in <-chan T) (_, _ <-chan T) {
```

Comment:
Whiy does this function need to be generic? The type we're sending over the update channels is always `event.GenericEvent` right?

Reply:
It doesn't have to be generic, I just thought it would be nicer this way.
As you said, we're only using it for `event.GenericEvent` so if you want I can switch it up to the concrete type, please let me know what you prefer

Reply:
I would prefer the concrete type unless @shaneutt thinks otherwise :) 

Generics are super super cool but I'd expect we'll end up with alot of Go newbies on this project

Reply:
I don't feel strongly about this, @astoycos seems to feel a little bit more strongly about it than I do, so unless you're feeling very strongly it should remain generic I would say let's just make it explicit like Andrew suggests since there's really no downside of that.

Reply:
Sure, no problem